azure-iot-hub/tests/test_job_manager.py

```diff
@@ -0,0 +1,174 @@
+# --------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+import pytest
+from azure.iot.hub.protocol.models import AuthenticationMechanism
+from azure.iot.hub.iothub_job_manager import IoTHubJobManager
+
+"""---Constants---"""
+
+fake_hostname = "beauxbatons.academy-net"
+fake_device_id = "MyPensieve"
+fake_shared_access_key_name = "alohomora"
+fake_shared_access_key = "Zm9vYmFy"
+fake_job_properties = "fake_job_properties"
+fake_job_id = "fake_job_id"
+fake_job_request = "fake_job_request"
+fake_job_type = "fake_job_type"
+fake_job_status = "fake_job_status"
+
+
+"""----Shared fixtures----"""
+
+
+@pytest.fixture(scope="function", autouse=True)
+def mock_job_client_operations(mocker):
+    mock_job_client_operations_init = mocker.patch(
+        "azure.iot.hub.protocol.iot_hub_gateway_service_ap_is.JobClientOperations"
+    )
+    return mock_job_client_operations_init.return_value
+
+
+@pytest.fixture(scope="function")
+def iothub_job_manager():
+    connection_string = "HostName={hostname};DeviceId={device_id};SharedAccessKeyName={skn};SharedAccessKey={sk}".format(
+        hostname=fake_hostname,
+        device_id=fake_device_id,
+        skn=fake_shared_access_key_name,
+        sk=fake_shared_access_key,
+    )
+    iothub_job_manager = IoTHubJobManager(connection_string)
+    return iothub_job_manager
+
+
+@pytest.mark.describe("IoTHubJobManager")
+class TestJobManager(object):
+    @pytest.mark.it("Instantiates with an empty connection string")
+    def test_instantiates_with_empty_connection_string(self):
+        with pytest.raises(Exception):
```

Comment:
>Exception [](start = 27, length = 9)

Never do this in pytests. Capture the SPECIFIC exception being raised. Capturing a generic Exception will catch anything, including unrelated errors, and actual failures in testing. We've had this problem a lot in device client, where many bugs were missed due to this exact mistake. Here and everywhere.